Refresher/Patching/Patcher.cs

```diff
@@ -16,79 +19,92 @@ public Patcher(Stream stream)
 
         this.Stream = stream;
 
-        this._targets = new Lazy<List<PatchTargetInfo>>(() => FindUrl(stream));
+        this.Stream.Position = 0;
+
+        this._targets = new(() => FindPatchables(stream));
     }
 
     public Stream Stream { get; }
 
-    [GeneratedRegex("^https?[^\\x00]\\/\\/([0-9a-zA-Z.:].*)\\/?([0-9a-zA-Z_]*)$", RegexOptions.Compiled)]
+    [GeneratedRegex("^https?[^\\x00]//([0-9a-zA-Z.:].*)/?([0-9a-zA-Z_]*)$", RegexOptions.Compiled)]
     private static partial Regex UrlMatch();
 
-    private static List<PatchTargetInfo> FindUrl(Stream file)
+    [GeneratedRegex("[a-zA-Z0-9!@#$%^&*()?/<>]", RegexOptions.Compiled)]
+    private static partial Regex DigestMatch();
+
+    /// <summary>
+    /// Finds a set of URLs and Digest keys in the given file, excluding C printf strings.
+    /// </summary>
+    /// <param name="file">A seekable stream containing the file to look through</param>
+    /// <returns>A list of the URLs and Digest keys</returns>
+    private static (List<PatchTargetInfo> urls, List<PatchTargetInfo> digests) FindPatchables(Stream file)
```

Comment:
```suggestion
    private static (List<PatchTargetInfo> urls, List<PatchTargetInfo> digests) FindPatchableElements(Stream file)
```

I also don't like this system of returning multiple lists, better to have one list and just have a Type on `PatchTargetInfo`